app/controllers/reservations_controller.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+class ReservationsController < ApplicationController
+  include JSONAPI::Fetching
+
+  def index
+    reservations = policy_scope(Reservation).where(screening: params[:screening_id])
+
+    render jsonapi: reservations, except: blacklisted_attributes
+  end
+
+  def show
+    reservation = policy_scope(Reservation).find(params[:id])
+
+    render jsonapi: reservation, except: blacklisted_attributes
+  end
+
+  def create
+    if already_booked?
+      render json: { error: 'One or more seats is taken' }, status: :unprocessable_entity
+    else
+      reservation = current_user.reservations.new(reservation_params)
+      authorize reservation
```

Comment:
So in the case of creating it is, in most cases, better to have `authorize Reservation` at the beginning of the function instead of authorizing whenever someone is able to create this exact resource.